### ISSUE TITLE
irodsFs returns ENOENT when an ires is down

### ISSUE THREAD

**Issue by mcast** [irods/irods#2710]
Please can those errors triggered by absence of a host[1] give a different errno, such as `EHOSTUNREACH`, `EHOSTDOWN` or the more vague `EREMOTEIO` ? 

This puzzled me for a while - I know I put the file there, and `ls` says it is still there, but when I try to read the file it isn't.

```
mca@eirods-dev:~$ irodsEnvFile=~/.irods/irodsenv-test  FuseCacheDir=/tmp/mca.fuseCache.test irodsFs ~/mnt
mca@eirods-dev:~$ ls -l mnt/_live_tmp/nfs/cancer_trk0033/0000xxxxxxxx723.tophat.nodup.bai; sha256sum /nfs/users/nfs_m/mca/mnt/_live_tmp/nfs/cancer_trk0033/0000xxxxxxxx723.tophat.nodup.bai
-rw-rw---- 1 mca cancer 7277400 Mar 31 17:45 /nfs/users/nfs_m/mca/mnt/_live_tmp/nfs/cancer_trk0033/0000xxxxxxxx723.tophat.nodup.bai
sha256sum: /nfs/users/nfs_m/mca/mnt/_live_tmp/nfs/cancer_trk0033/0000xxxxxxxx723.tophat.nodup.bai: No such file or directory
1@eirods-dev:~$ irodsFs --version
Version: rods4.0.3mca@eirods-dev:~$ 
```

(It is talking to a v3.3.1 server, but I think that isn't relevant at this point.  It gives the same failure as the 3.3.1 irodsFs+server, and I wanted to report the irodsFs problem against the maintained branch.  Other 4.0.3 icommands fail with `CAT_UNKNOWN_TABLE` against 3.3.1 server.)

Thanks,

---

[1] In this case,

```
mca@irods-cgp-dev:~$ export irodsEnvFile=~/.irods/irodsenv-test
mca@irods-cgp-dev:~$ ils -L _live_tmp/nfs/cancer_trk0033/0000xxxxxxxx723.tophat.nodup.bai
  filetrk           0 nxxxxxxrk0033             7277400 2015-03-31.17:45 & 545723.tophat.nodup.bai
            generic    /nfs/cancer_trk0033/0000xxxxxxxx723.tophat.nodup.bai
mca@irods-cgp-dev:~$ irodsEnvFile=~/.irods/irodsenv-test ilsresc -l nxxxxxxrk0033 | grep locat
location: cgp-1-1-02.internal.sanger.ac.uk
mca@irods-cgp-dev:~$ telnet cgp-1-1-02.internal.sanger.ac.uk 1247
Trying 172.17.140.33...
telnet: Unable to connect to remote host: Connection refused
```

that iRes is down for maintenance.  Those are v3.3.1 icommands on the 3.3.1 server.

**Comment by trel** [comment]
Thanks @mcast, I'm not sure we'll get to looking at this real soon (4.1.0 coming out this week and user group meeting a week later), but we can add this scenario to our tests when we have some moments.

**Comment by trel** [comment]
@iychoi @xu-hao is this still an issue?  can this be closed?

**Comment by iychoi** [comment]
Well, current irodsFs is based on the old irodsFs. Basically it is programmed to return an -ENOENT error or error code from underlying iRODS client C API.
I think adding a quick connection test when the irodsFs is started can return error messages in console when iCAT server is not reachable.
I'll fix it soon. Let's open this for a while.

**Comment by iychoi** [comment]
I just made a pull request #2877 which solves this issue. In case of misconfiguring irods environment or host down, it will immediately return and print messages.

**Comment by trel** [comment]
this was closed via #2877
